app/statistics/controller.js

```diff
@@ -266,15 +190,63 @@ export default class StatisticsController extends Controller.extend(
     });
   }
 
+  @action
+  changeFromDate(date) {
+    this.fromDate = serializeMoment(date);
+    this.data.perform();
+  }
+
+  get getFromDate() {
+    return deserializeMoment(this.fromDate);
+  }
+
+  @action
+  changeToDate(date) {
+    this.toDate = serializeMoment(date);
+    this.data.perform();
+  }
+
+  get getToDate() {
+    return deserializeMoment(this.toDate);
+  }
+
   @action
   setModelFilter(key, value) {
     set(this, key, value && value.id);
+    this.data.perform();
   }
 
   @action
   resetQP() {
     this.resetQueryParams(
-      Object.keys(this.allQueryParams).filter((qp) => qp !== "type")
+      Object.keys(allQueryParams(this)).filter((qp) => qp !== "type")
     );
   }
+
+  get queryParamsState() {
+    const states = {};
+    for (const param of this.queryParams) {
+      const defaultValue = getDefaultQueryParamValue(param);
+      const currentValue = this[param];
+      states[param] = {
+        as: param,
```

Comment:
the `as` is an parachute artifact

Reply:
i was making sure to make exact replacement of ember parachute
i remember that u ask me before about why it's returning `undefined`, that's because ember parachute was returning that too